auth0/resource_auth0_attack_protection.go

```diff
@@ -409,111 +393,169 @@ func updateAttackProtection(ctx context.Context, d *schema.ResourceData, m inter
 }
 
 func flattenBreachedPasswordProtection(bpd *management.BreachedPasswordDetection) []interface{} {
-	m := make(map[string]interface{})
-	if bpd != nil {
-		m["enabled"] = bpd.Enabled
-		m["admin_notification_frequency"] = bpd.AdminNotificationFrequency
-		m["method"] = bpd.Method
-		m["shields"] = bpd.Shields
+	var adminNotificationFrequency []interface{}
+	for _, frequency := range bpd.GetAdminNotificationFrequency() {
+		adminNotificationFrequency = append(adminNotificationFrequency, frequency)
+	}
+
+	var shields []interface{}
+	for _, shield := range bpd.GetShields() {
+		shields = append(shields, shield)
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"enabled":                      bpd.GetEnabled(),
+			"method":                       bpd.GetMethod(),
+			"admin_notification_frequency": adminNotificationFrequency,
+			"shields":                      shields,
+		},
 	}
-	return []interface{}{m}
 }
 
 func flattenBotDetection(bot *management.BotDetection) []interface{} {
-	m := make(map[string]interface{})
-
-	m["response"] = []interface{}{
+	return []interface{}{
 		map[string]interface{}{
-			"policy":   bot.GetResponse().GetPolicy(),
-			"selected": bot.GetResponse().GetSelected(),
-			"providers": []interface{}{
+			"response": []interface{}{
 				map[string]interface{}{
-					"recaptcha_v2": []interface{}{
+					"policy":   bot.GetResponse().GetPolicy(),
+					"selected": bot.GetResponse().GetSelected(),
+					"providers": []interface{}{
 						map[string]interface{}{
-							"secret":   bot.GetResponse().GetProviders().GetRecaptchaV2().GetSecret(),
-							"site_key": bot.GetResponse().GetProviders().GetRecaptchaV2().GetSiteKey(),
-						},
-					},
-					"recaptcha_enterprise": []interface{}{
-						map[string]interface{}{
-							"api_key":    bot.GetResponse().GetProviders().GetRecaptchaEnterprise().GetAPIKey(),
-							"project_id": bot.GetResponse().GetProviders().GetRecaptchaEnterprise().GetProjectID(),
-							"site_key":   bot.GetResponse().GetProviders().GetRecaptchaEnterprise().GetSiteKey(),
+							"recaptcha_v2": []interface{}{
+								map[string]interface{}{
+									"secret":   bot.GetResponse().GetProviders().GetRecaptchaV2().GetSecret(),
+									"site_key": bot.GetResponse().GetProviders().GetRecaptchaV2().GetSiteKey(),
+								},
+							},
+							"recaptcha_enterprise": []interface{}{
+								map[string]interface{}{
+									"api_key":    bot.GetResponse().GetProviders().GetRecaptchaEnterprise().GetAPIKey(),
+									"project_id": bot.GetResponse().GetProviders().GetRecaptchaEnterprise().GetProjectID(),
+									"site_key":   bot.GetResponse().GetProviders().GetRecaptchaEnterprise().GetSiteKey(),
+								},
+							},
 						},
 					},
 				},
 			},
 		},
 	}
-
-	return []interface{}{m}
 }
 
 func expandSuspiciousIPThrottling(d *schema.ResourceData) *management.SuspiciousIPThrottling {
-	var ipt management.SuspiciousIPThrottling
+	var ipt *management.SuspiciousIPThrottling
```

Comment:
This should always be a pointer so the checks in the update to see if ipt is nil or not, actually work and skip a request if it's unnecessary.